File: Program's_Contributed_By_Contributors/Java_Programs/Searches/BinarySearch.java
'''
Binary Search in Java
Author: Rishabh Rathore
'''
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr ={-18,-10,-7,-2,0,4,45,55,65,75,79,99};
        int target = 4;
        int ans = binarySearch(arr,target ) ;
        System.out.println(ans);

    }
    static int binarySearch(int[] arr,int target){
        int start = 0;
        int end = arr.length - 1;
        while(start <= end){
            //int mid = (start + end) /2;
            int mid = start + (end - start) /2;
            if(target < arr[mid]){
                end = mid - 1;
            }
            else if (target > arr[mid]){
                start = mid + 1;
            }
            else {
                return mid;
            }
        }
        return -1;
    }
}

File: Program's_Contributed_By_Contributors/C++ Programs/Searching/binarySearch.cpp
// Binary Search
#include<iostream>
using namespace std;

int binarySearch(int arr[], int n, int key){
    int s=0;
    int e=n;

    while(s<=e){
        int mid = (s+e)/2;
        if (arr[mid]==key){
            return mid;
        }
        else if (arr[mid]>key){
            e = mid-1;
        }
        else{
            s = mid+1;
        }
        
    }return -1;
}

int main(int argc, char const *argv[])
{
    int n;
    cin>>n;
    int array[n];

    for (int i=0; i<n;i++){
        cin>>array[i];
    }

    int key;
    cin>>key;

    cout<< binarySearch(array, n, key)<<endl;

    return 0;
}